diff --git a/Makefile.am b/Makefile.am
index 7b281d7..9f78ff8 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -3,7 +3,14 @@
 AUTOMAKE_OPTIONS = foreign 1.4
 ACLOCAL_AMFLAGS = -I m4
 
-SUBDIRS = Solver
+SUBDIRS = Solver tests
+
+profile:
+	make clean
+	make CXXFLAGS="$(CXXFLAGS) -fprofile-generate" -j8
+	make check CXXFLAGS="$(CXXFLAGS) -fprofile-generate"
+	make mostlyclean
+	make CXXFLAGS="$(CXXFLAGS) -fprofile-use" -j8
 
 all-local: Solver
 	cp Solver/cryptominisat .
diff --git a/configure.in b/configure.in
index 5b539a1..40b0460 100644
--- a/configure.in
+++ b/configure.in
@@ -11,7 +11,7 @@ if test -n "${CXXFLAGS}"; then
 fi
 AC_PROG_CXX
 if test X$user_set_cxxflags != Xyes; then
-  CXXFLAGS="-g -O2"
+  CXXFLAGS="-g -O3"
 fi
 AC_LANG(C++)
 
@@ -160,4 +160,4 @@ if test "$use_static" = "yes"; then
   LDFLAGS="$LDFLAGS -all-static"
 fi
 
-AC_OUTPUT(Makefile Solver/Makefile)
+AC_OUTPUT(Makefile Solver/Makefile tests/Makefile)
diff --git a/tests/Makefile.am b/tests/Makefile.am
new file mode 100644
index 0000000..7d042f7
--- /dev/null
+++ b/tests/Makefile.am
@@ -0,0 +1,8 @@
+METASOURCES = AUTO
+
+#TESTS = cryptominisat
+
+check-local:
+	cp $(buildddir)../cryptominisat .
+	chmod +x $(srcdir)/test.sh
+	$(srcdir)/test.sh
diff --git a/tests/test.sh b/tests/test.sh
new file mode 100755
index 0000000..b4e3526
--- /dev/null
+++ b/tests/test.sh
@@ -0,0 +1,6 @@
+#!/bin/bash
+myret="`./cryptominisat --nosolprint --verbosity=1 AProVE09-12.cnf.gz`"
+if [ "$myret" == "c SATISFIABLE" ]
+then
+exit 0
+fi
